bin/run.sh

```diff
@@ -58,7 +60,12 @@ set +e
 # Run the tests for the provided implementation file and redirect stdout and
 # stderr to capture it
 test_output=$(stack build --resolver lts-20.18 --test --allow-different-user 2>&1)
-cp "${input_dir}/results.json" "${results_file}"
+
+# Copy results.json to the output directory (only if output directory is different from
+# the input directory)
+if [ "${input_dir}/results.json" != "${results_file}" ]; then
+  mv "${input_dir}/results.json" "${results_file}"
+fi
```

Comment:
Why this change?

Reply:
The tests were failing for me when run with `bin/run-tests-in-docker.sh` or even `bin/run-tests.sh` because `cp` (or `mv`) throws an error saying `'tests/example-all-fail/results.json' and 'tests/example-all-fail/results.json' are the same file`.

This if statement guards against this case where the input_dir and output_dir are the same.

Reply:
And without this `mv` command the file does not end up in the `output_dir` as the hspec formatter outputs it inside `input_dir`